cmd/chain-maind/app/app.go

```diff
@@ -150,13 +167,27 @@ func initRootCmd(rootCmd *cobra.Command, encodingConfig params.EncodingConfig) {
 		if err := mergo.Merge(&genesis, genesisPatch, mergo.WithOverride); err != nil {
 			return err
 		}
-		if err := file.Truncate(0); err != nil {
+
+		jsonEncoder := json.NewEncoder(tmpFile)
```

Comment:
how about combine below lines in to one function 
for simplicity?

Reply:
Has been rewritten in another way.